File: ui/hospital-system-ui/src/types/tender-commission.ts
import { CommissionRole } from './enums'

export interface TenderCommissionResponse {
  id: number
  tenderId: number
  userId: number
  userFullName: string
  userTitle: string | null
  role: CommissionRole
  createdAt: string
}

export interface CreateTenderCommissionRequest {
  tenderId: number
  userId: number
  role: CommissionRole
}

File: server/src/main/java/com/hospitaltender/server/entity/TenderCommission.java
package com.hospitaltender.server.entity;

import com.hospitaltender.server.enums.CommissionRole;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Entity
@Table(name = "tender_commissions")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class TenderCommission extends BaseEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "tender_id", nullable = false)
    private Tender tender;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    @Column(name = "role", length = 50)
    private CommissionRole role;
}
